synapse/storage/data_stores/main/search.py

```diff
@@ -693,6 +704,21 @@ def _to_postgres_options(options_dict):
     return "'%s'" % (",".join("%s=%s" % (k, v) for k, v in options_dict.items()),)
 
 
+def _extract_search_keywords(entry: str) -> Iterable[str]:
+    """
+    Extract an iterator of words from the string.
+
+    For instance: https://github.com/matrix-org/synapse/issues/3024
+    becomes: ["https","github","com","matrix-org","synapse","issues","3024"].
+
+    Args:
+        entry: A string to be pre-processed for searching.
+
+    Returns: the words in the string.
+    """
+    return re.finditer(r"([\w\-]+)", entry.value)
```

Comment:
wait, if `entry` is a str, it doesn't have a `value` property.

Also: `finditer` returns `Match` objects, not `str`s. You'll either need to extract the matched string from the Match objects (`m[0] for m in matches`), or go back to `findall`.